vignettes/climr_workflow_beg.Rmd

```diff
@@ -40,64 +40,89 @@ cpath <- if (interactive()) {
+  max_run = 3, # specify 3 individual runs for each model
   vars = c("PPT", "CMD", "CMI")
 )
 ```
 
-These functions are useful to list available options:
+The resulting output is a data.table, printed below:
+
+```{r, echo = FALSE, results = "asis"}
+knitr::kable(ds_out[id == head(id, 1)], caption = "'ds_out' table - Output from `climr_downscale`. Outputs are shown for one location ('id')")
+```
+
+###Listing available options
 
-```{r, eval = FALSE}
+The `list_*` functions (see ?list_gcm) are useful to see available options, for example:
+
+```{r, eval = TRUE}
 list_gcm()
 list_gcm_period()
 list_ssp()
-list_variables()
 ```
 
-```{r, echo = FALSE, results = "asis"}
-knitr::kable(ds_out[id == head(id, 1)], caption = "'ds_out' table - Output from `climr_downscale`. Outputs are shown for one location ('id')")
-```
+These functions can be used as arguments in `climr_downscale()`. For example, `vars = list_variables()` could be specified to obtain data for all available variables, or `gcm_models = list_gcm()[c(1,4,5,6,7,10,11,12)]` could be used to specify the eight-gcm ensemble recommended by Mahony et al. (2022). 
 
-Alternatively, you may want to download and investigate the normals and anomalies
-before downscaling. See the section "Workflow with `*_input` functions and `downscale`"
-in `vignette("vignettes/climr_workflow_int.Rmd")` for an example.
+###Working with raster data
+A common use of `climr` is to obtain climate data for a user-provided raster grid. Here is some sample code showing how to do so. We convert a digital elevation model (DEM) into a table of points (one per grid cell) which we use as the `climr` input table. Then we take the `climr` input and populate the raster grid with a climate variable. 
 
-#### Adding / Modifying climate variables
+```{r, eval = TRUE}
+library(terra)
+library(data.table)
 
-All climate variables are computed via [R/append_clim_vars.R](./R/append_clim_vars.R).
-Add or modify entries in the variables list as needed.
-Use function `v` to handle dependencies (i.e. `v("Tmin")` instead of `dt$Tmin` / `dt[["Tmin"]]`).
+## get the sample digital elevation model (dem) provided with `climr`
+dem_vancouver <- get(data("dem_vancouver")) |> unwrap()
 
-#### Lapse rates details
+## convert the DEM to a data.frame
+my_grid <- as.data.frame(dem_vancouver, cells=T, xy=T)
+colnames(my_grid) <- c("id", "lon", "lat", "elev") # rename column names to what climr expects
 
-See `vignette("vignettes/lapse_rates.Rmd")` for details.
+## A simple climr query. This will return the observed 1961-1990 and 2001-2020 mean annual temperature (MAT) for the raster grid points. 
+ds_out <- climr_downscale(
+  xyz = my_grid, 
+  historic_period = "2001_2020", 
+  vars = c("MAT")
+)
 
-#### Use of terra package
+## populate the raster grid with the downscaled climate values
+my_clim <- dem_vancouver # use the DEM as a template raster
+values(my_clim) <- NA # remove the elevation values from the 
```

Comment:
instead of this, you can replace the line above by:
`my_clim <- rast(dem_vancouver) `, which creates an empty raster with the same attributes as `dem_vancouver`

Reply:
Nice!